File: packages/python/evalforge/scorers/exact.py
from ..models import EvalCase, EvalResult
from .base import Scorer


class ExactMatchScorer(Scorer):
    name = "exact_match"

    def __init__(self, case_sensitive: bool = False, strip: bool = True):
        self.case_sensitive = case_sensitive
        self.strip          = strip

    def score(self, case: EvalCase, output: str) -> EvalResult:
        expected = case.expected or ""
        a, b = output, expected
        if self.strip:
            a, b = a.strip(), b.strip()
        if not self.case_sensitive:
            a, b = a.lower(), b.lower()
        match = a == b
        return self._result(case, output, 1.0 if match else 0.0, reason=None)

File: packages/typescript/src/scorers/exact.ts
import { EvalCase, EvalResult } from "../models";
import { Scorer, makeResult } from "./base";

export class ExactMatchScorer implements Scorer {
  readonly name = "exact_match";

  constructor(
    private caseSensitive: boolean = false,
    private strip:         boolean = true,
  ) {}

  score(evalCase: EvalCase, output: string): EvalResult {
    let a = output;
    let b = evalCase.expected ?? "";
    if (this.strip) { a = a.trim(); b = b.trim(); }
    if (!this.caseSensitive) { a = a.toLowerCase(); b = b.toLowerCase(); }
    return makeResult(evalCase, output, a === b ? 1.0 : 0.0);
  }
}
